Description: Fix incorrect handling of comments when the file starts with
 a comment, fix failure when parsing a polygon starting with a negative
 coordinate.
Origin: http://bazaar.launchpad.net/~scouring/scour/trunk/revision/216
---
 scour.py |   11 +++++++----
 1 file changed, 7 insertions(+), 4 deletions(-)

--- scour.orig/scour.py
+++ scour/scour.py
@@ -2125,10 +2125,13 @@ def parseListOfPoints(s):
 		# we got negative coords
 		else:
 			for j in xrange(len(negcoords)):
-				# first number could be positive
 				if j == 0:
+					# first number could be positive
 					if negcoords[0] != '':
 						nums.append(negcoords[0])
+				# but it could also be negative
+				elif len(nums) == 0:
+					nums.append('-' + negcoords[j])
 				# otherwise all other strings will be negative
 				else:
 					# unless we accidentally split a number that was in scientific notation
@@ -2554,9 +2557,9 @@ def removeComments(element) :
 		# must process the document object separately, because its
 		# documentElement's nodes have None as their parentNode
 		for subelement in element.childNodes:
-			if isinstance(element, xml.dom.minidom.Comment):
-				numCommentBytes += len(element.data)
-				element.documentElement.removeChild(subelement)
+			if isinstance(subelement, xml.dom.minidom.Comment):
+				numCommentBytes += len(subelement.data)
+				element.removeChild(subelement)
 			else:
 				removeComments(subelement)
 	elif isinstance(element, xml.dom.minidom.Comment):
